app/src/main/java/com/keylesspalace/tusky/components/account/list/ListSelectionFragment.kt

```diff
@@ -16,90 +16,116 @@
 
 package com.keylesspalace.tusky.components.account.list
 
+import android.app.Dialog
+import android.content.Context
+import android.content.Intent
 import android.os.Bundle
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.LinearLayout
+import androidx.appcompat.app.AlertDialog
 import androidx.fragment.app.DialogFragment
 import androidx.fragment.app.viewModels
 import androidx.lifecycle.lifecycleScope
 import androidx.recyclerview.widget.DiffUtil
-import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.ListAdapter
 import com.google.android.material.snackbar.Snackbar
+import com.keylesspalace.tusky.ListsActivity
 import com.keylesspalace.tusky.R
-import com.keylesspalace.tusky.databinding.FragmentListsForAccountBinding
-import com.keylesspalace.tusky.databinding.ItemAddOrRemoveFromListBinding
+import com.keylesspalace.tusky.databinding.FragmentListsListBinding
+import com.keylesspalace.tusky.databinding.ItemListBinding
 import com.keylesspalace.tusky.di.Injectable
 import com.keylesspalace.tusky.di.ViewModelFactory
+import com.keylesspalace.tusky.entity.MastoList
 import com.keylesspalace.tusky.util.BindingHolder
 import com.keylesspalace.tusky.util.hide
 import com.keylesspalace.tusky.util.show
-import com.keylesspalace.tusky.util.viewBinding
 import com.keylesspalace.tusky.util.visible
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.awaitCancellation
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.launch
 import java.io.IOException
 import javax.inject.Inject
 
-class ListsForAccountFragment : DialogFragment(), Injectable {
+class ListSelectionFragment : DialogFragment(), Injectable {
+
+    interface ListSelectionListener {
+        fun onListSelected(list: MastoList)
+    }
 
     @Inject
     lateinit var viewModelFactory: ViewModelFactory
 
     private val viewModel: ListsForAccountViewModel by viewModels { viewModelFactory }
-    private val binding by viewBinding(FragmentListsForAccountBinding::bind)
+
+    private var _binding: FragmentListsListBinding? = null
+
+    // This property is only valid between onCreateDialog and onDestroyView
+    private val binding get() = _binding!!
 
     private val adapter = Adapter()
 
+    private var selectListener: ListSelectionListener? =  null
+    private var accountId: String? = null
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        selectListener = context as? ListSelectionListener
```

Comment:
I like this implicit configurability: If the caller implements the interface it will be informed of a selection - otherwise nothing happens.